src/runmode-dpdk.c

```diff
@@ -0,0 +1,1295 @@
+        if (retval != 0) {
+            SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) during obtaining port id of %s", retval,
+                    iconf->out_iface);
+            SCReturnInt(retval);
+        }
+
+        if (rte_eth_dev_socket_id(iconf->port_id) != rte_eth_dev_socket_id(iconf->out_port_id)) {
+            SCLogWarning(SC_WARN_DPDK_CONF, "%s and %s are not on the same NUMA node", iconf->iface,
+                    iconf->out_iface);
+        }
+
+        retval = DeviceValidateOutIfaceConfig(iconf);
+        if (retval != 0) {
+            // Error will be written out by the validation function
+            SCReturnInt(retval);
+        }
+    }
+    SCReturnInt(0);
+}
+
+int DeviceConfigure(DPDKIfaceConfig *iconf)
+{
+    // configure device
+    int retval;
+    struct rte_eth_dev_info dev_info;
+    struct rte_eth_conf port_conf;
+
+    retval = rte_eth_dev_get_port_by_name(iconf->iface, &(iconf->port_id));
+    if (retval < 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) when getting port id of %s Is device enabled?",
+                retval, iconf->iface);
+        SCReturnInt(retval);
+    }
+
+    if (!rte_eth_dev_is_valid_port(iconf->port_id)) {
+        SCLogError(SC_ERR_DPDK_INIT, "Specified port %d is invalid", iconf->port_id);
+        SCReturnInt(retval);
+    }
+
+    retval = rte_eth_dev_socket_id(iconf->port_id);
+    if (retval < 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) invalid socket id (port %u)", retval,
+                iconf->port_id);
+        SCReturnInt(retval);
+    } else {
+        iconf->socket_id = retval;
+    }
+
+    retval = rte_eth_dev_info_get(iconf->port_id, &dev_info);
+    if (retval != 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) during getting device info (port %" PRIu16 ")",
+                retval, iconf->port_id);
+        SCReturnInt(retval);
+    }
+
+    if (iconf->nb_rx_queues > dev_info.max_rx_queues) {
+        errno = ERANGE;
+        SCLogError(SC_ERR_DPDK_INIT,
+                "Number of configured RX queues of port %" PRIu16
+                " is higher than maximum allowed (%" PRIu16 ")",
+                iconf->port_id, dev_info.max_rx_queues);
+        SCReturnInt(-errno);
+    }
+
+    if (iconf->nb_tx_queues > dev_info.max_tx_queues) {
+        errno = ERANGE;
+        SCLogError(SC_ERR_DPDK_INIT,
+                "Number of configured TX queues of port %" PRIu16
+                " is higher than maximum allowed (%" PRIu16 ")",
+                iconf->port_id, dev_info.max_tx_queues);
+        SCReturnInt(-errno);
+    }
+
+    if (iconf->mtu > dev_info.max_mtu || iconf->mtu < dev_info.min_mtu) {
+        errno = ERANGE;
+        SCLogError(SC_ERR_DPDK_INIT,
+                "Loaded MTU of port %" PRIu16 " is out of bounds. "
+                "Min MTU: %" PRIu16 " Max MTU: %" PRIu16,
+                iconf->port_id, dev_info.min_mtu, dev_info.max_mtu);
+        SCReturnInt(-errno);
+    }
```

Comment:
As mentioned in above comment the default port_conf is set to 1500 Byte, even if Jumbo is supported there is no change done to reflect the RX side for large packet support. Also there is no check done for mempool needs to created for max size as some nic does not support segmented rx or tx. please rerwork.